static/app/components/events/autofix/autofixSteps.tsx

```diff
@@ -167,6 +179,20 @@ export function AutofixSteps({data, groupId, runId}: AutofixStepsProps) {
     replaceHeadersWithBold(logs.at(-1)?.message ?? '') ??
     '';
 
+  const hasRootCauseStep = steps.some(
+    step => step.type === AutofixStepType.ROOT_CAUSE_ANALYSIS
+  );
+  const hasSolutionStep = steps.some(step => step.type === AutofixStepType.SOLUTION);
+  const hasChangesStep = steps.some(step => step.type === AutofixStepType.CHANGES);
+
+  const isRootCauseFirstAppearance = hasRootCauseStep && !previousRootCauseRef.current;
+  const isSolutionFirstAppearance = hasSolutionStep && !previousSolutionRef.current;
+  const isChangesFirstAppearance = hasChangesStep && !previousChangesRef.current;
+
+  previousRootCauseRef.current = hasRootCauseStep;
+  previousSolutionRef.current = hasSolutionStep;
+  previousChangesRef.current = hasChangesStep;
```

Comment:
can't we achieve the same but in a much simpler way by just maintaining an `isMounted` (or something similar if we need to wait for data to load) state for the whole drawer/all the steps, and anything that appears after initial mount is just considered that step being polled in?

Reply:
ok